lib/Frontend/CachingUtils.cpp

```diff
@@ -188,6 +188,22 @@ bool replayCachedCompilerOutputs(
           if (!Proxy)
             return Proxy.takeError();
 
+          if (Kind == file_types::ID::TY_Object && UseCASBackend) {
```

Comment:
Sorry, I missed this from the previous review. You should not write here. The replay utils does two steps:
step 1, make sure all outputs can be found
step 2, write outputs only if all outputs can be found, otherwise, do the regular compilation.

You should not try to write one output before confirming all outputs can be found.

Reply:
So what I had before was correct, just use an OS and then write to *File when everything is resolved

Reply:
I would rather you materialize the object afterwards as well. We don't expect the cache lookup succeeded but materialize object failed (and you are going to error out anyway), so just move the materialization of object file from MCCAS to afterwards.